router/event.go

```diff
@@ -103,6 +106,28 @@ func (router *Router) eventFromRequest(r *http.Request) (*eventpkg.Event, string
 	return event, path, nil
 }
 
+func (router *Router) parseCustomEventAsCloudEvent(originalEventType eventpkg.Type, contentType string, body []byte) (*eventpkg.Event, error) {
+	var event = &eventpkg.Event{}
+	if originalEventType == eventpkg.TypeHTTP || originalEventType == eventpkg.TypeInvoke {
+		return event, errors.New("not a custom event")
+	}
+
+	err := json.Unmarshal(body, event)
+	if err != nil {
+		return event, err
+	}
+	if len(event.EventType) < 0 ||
```

Comment:
It's better to use validation tags and validate the event with validator used in other places.